File: sdk/java/src/main/java/vn/hitechcloud/sdk/resources/Pmg.java
package vn.hitechcloud.sdk.resources;

import java.util.Map;
import vn.hitechcloud.sdk.HttpClient;

public class Pmg extends BaseResource {

    public Pmg(HttpClient http) {
        super(http);
    }

    /** Get config */
    public Map<String, Object> config(int id) throws Exception {
        return http.get(String.format("/api/service/%id/htcpmg/config", id));
    }

    /** Add domain */
    public Map<String, Object> addDomain(int id, Map<String, Object> data) throws Exception {
        return http.post(String.format("/api/service/%id/htcpmg/domains", id), data);
    }

    /** Add transport */
    public Map<String, Object> addTransport(int id, Map<String, Object> data) throws Exception {
        return http.post(String.format("/api/service/%id/htcpmg/transport", id), data);
    }

    /** Get stats */
    public Map<String, Object> stats(int id) throws Exception {
        return http.get(String.format("/api/service/%id/htcpmg/stats", id));
    }

}

File: sdk/nodejs/src/resources/pmg.js
"use strict";

const { BaseResource } = require("./base");

class PmgResource extends BaseResource {
  async getConfig(serviceId) {
    return this._http.get("/api/service/{id}/htcpmg/config");
  }

  async addDomain(serviceId, data) {
    return this._http.post("/api/service/{id}/htcpmg/domains", data);
  }

  async setTransport(serviceId, data) {
    return this._http.post("/api/service/{id}/htcpmg/transport", data);
  }

  async getStats(serviceId) {
    return this._http.get("/api/service/{id}/htcpmg/stats");
  }

}

module.exports = { PmgResource };
